cmd/bnscli/cmd_gov.go

```diff
@@ -67,6 +59,46 @@ transaction (ie signatures) are being dropped.
 		option.Option = &app.ProposalOptions_ResetRevenueMsg{
 			ResetRevenueMsg: msg,
 		}
+	case *app.BatchMsg:
+		msgs, err := msg.MsgList()
+		if err != nil {
+			return fmt.Errorf("cannot extract messages: %s", err)
+		}
+		var messages []app.ProposalBatchMsg_Union
+		for _, m := range msgs {
+			switch m := m.(type) {
+			case *cash.SendMsg:
```

Comment:
Same point I raised about batch (please as a separate PR).
Since people will not look at this logic when adding a new type, it would be good to add support for all supported types proposal batch. (Both in this case, and above in the top level case).

This logic here is pretty repetative and the type-checker will ensure you make no mistakes, so I don't think it matters if some of those cases are not tested

Reply:
Done in https://github.com/iov-one/weave/pull/735/commits/0804afabf48a35413a0603c97d1cbbf065165380 and https://github.com/iov-one/weave/pull/735/commits/b36988931519e7ffdc52426b756b86d580383ae7